pkg/envoy/eds/builder.go

```diff
@@ -17,6 +18,19 @@ const (
 	remoteClusterPriority = uint32(1)
 )
 
+type endpointsBuilder struct {
+	upstreamSvcEndpoints map[service.MeshService][]endpoint.Endpoint
+}
+
+func (b *endpointsBuilder) Build() []types.Resource {
+	var edsResources []types.Resource
+
+	for svc, endpoints := range b.upstreamSvcEndpoints {
+		edsResources = append(edsResources, newClusterLoadAssignment(svc, endpoints))
```

Comment:
it's preferrable (but not required), to not pass fields to methods that further operate on those fields. The reason being that it is much easier to see where the fields are used otherwise, in that with an IDE that leverages gopls, I can simply clickthrough that field to view all of its uses. 

When it gets passed around however, I then need to continue to manually drill into each function call.

This isn't so bad in this scenario, but there's instances in our code base where that cognitive overhead does add up.

One option here would be to remove unused parts of the code, and inline that function call here. You could get rid of lines 54-56 and 75-89 which are currently unused

Reply:
I prefer not removing these lines in this PR. Although they are not used now, but the features were previously completed. Removing them will break these features. We can remove or change them in separate PR.